itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosExternalTables.java

```diff
@@ -1937,4 +1937,84 @@ public void testSingleCopyTasksConfiguration(boolean runCopyTasksOnTarget)
     // Clean up the filter file.
     new File(filterFilePath).delete();
   }
+
+  @Test
+  public void testTableAndPartitionExportDumpWithParallelism() throws Throwable {
+    testTableExportDumpParallelism(true);
+  }
+
+  @Test
+  public void testTableAndPartitionExportDumpWithNoParallelism() throws Throwable {
+    testTableExportDumpParallelism(false);
+  }
+
+  public void testTableExportDumpParallelism(boolean enableParallelism) throws Throwable {
+    List<String> tableList = new ArrayList<String>();
+    List<String> withClause = ReplicationTestUtils.includeExternalTableClause(true);
+    if (enableParallelism) {
+      withClause.add("'" + HiveConf.ConfVars.REPL_EXPORT_DUMP_PARALLELISM.varname + "'='100'");
```

Comment:
Should also test where there are less threads than tables and partitions.
Especially the cases where the number of threads is less than the number of tables and number of partitions.
For instance 10 threads, 20 tables, 4th and 5th tables have > 10 partitions.